Application/EdFi.Ods.Api/Helpers/AssemblyLoaderHelper.cs

```diff
@@ -191,7 +203,38 @@ static bool HasPlugin(Assembly assembly)
             }
         }
 
-        private class PluginAssemblyLoadingContext : AssemblyLoadContext
+        private static bool IsExtensionAssembly(AssemblyMetadata assemblyMetadata)
+        {
+            return assemblyMetadata.AssemblyModelType.EqualsIgnoreCase(PluginConventions.Extension);
+        }
+
+        private static bool IsProfileAssembly(AssemblyMetadata assemblyMetadata)
+        {
+            return assemblyMetadata.AssemblyModelType.EqualsIgnoreCase(PluginConventions.Profile);
+        }
+
+        private static AssemblyMetadata LoadAssemblyMetadata(Assembly assembly)
+        {
+            var resourceName = assembly.GetManifestResourceNames()
+                .SingleOrDefault(p => p.EndsWith(AssemblyMetadataSearchString));
+            
+            if (resourceName == null)
+            {
+                throw new FileNotFoundException($"Assembly metadata file '{assembly.GetManifestResourceNames()}' not found for this project.");

```

Comment:
Usage of a `FileNotFoundException` is no longer appropriate here since we're looking for an embedded resource, not a file. Use `Exception` instead.

The parameter in the error message is now using `assembly.GetManifestResourceNames()` which is wrong on 2 accounts:

1. That method returns a `string[]`, which when embedded in the message like this will result in the literal text `System.String[]`, not the members of the array in a readable format.
2. The names of all the resources that are actually in the assembly isn't right for this message. We should be informing on which one is _missing_. This would be the value stored in the `AssemblyMetadataSearchString` constant.

Fix the message, and change the use of the term "project" to be "assembly", as that is what is actually be searched.